legate/core/operation.py

```diff
@@ -470,9 +611,38 @@ def get_requirement(
         tag = self.get_tag(strategy, part_symb)
         return req, tag, store_part
 
+
+class AutoTask(AutoOperation, Task):
+    def __init__(
+        self,
+        context: Context,
+        task_id: int,
+        mapper_id: int,
+        op_id: int,
+    ) -> None:
+        super().__init__(
+            context=context,
+            task_id=task_id,
+            mapper_id=mapper_id,
+            op_id=op_id,
+        )
+        self._reusable_stores: list[Tuple[Store, PartSym]] = []
+        self._reuse_map: dict[int, Store] = {}
+
     def add_input(
         self, store: Store, partition: Optional[PartSym] = None
     ) -> None:
+        """
+        Adds a store as input to the task
+
```

Comment:
Do we need the docstring in the subclass? @bryevdv I think Sphinx automatically adds the docstring if the parent class provides it? I'm not sure what the best way is to document abstract methods so that we don't have to repeat identical docstrings in each subclass.

Reply:
Not necessary for this PR. I would worry about this later.

Reply:
> Do we need the docstring in the subclass? 

No we don't in general, but look at their signatures. The methods are actually different and that was intentional, as the methods admit different types of arguments.

Reply:
Ah, okay.